server/src/main/java/org/loomdev/loom/entity/LivingEntityImpl.java

```diff
@@ -190,74 +189,18 @@ public boolean isHolding(@NotNull Predicate<ItemType> predicate) {
-        var mcStack = getMinecraftEntity().getItemInHand(hand == Hand.MAIN_HAND ? InteractionHand.MAIN_HAND : InteractionHand.OFF_HAND);
-
-        if (mcStack == net.minecraft.world.item.ItemStack.EMPTY) {
-            return Optional.empty();
-        }
-
-        return Optional.of(ItemStackImpl.of(mcStack));
-    }
-
-    @Override
-    public void setItemInHand(@NotNull Hand hand, @NotNull ItemStack itemStack) {
-        var mcStack = ((ItemStackImpl) itemStack).getMinecraftItemStack();
-        getMinecraftEntity().setItemInHand(hand == Hand.MAIN_HAND ? InteractionHand.MAIN_HAND : InteractionHand.OFF_HAND, mcStack);
-    }
-
-    @NotNull
-    private Optional<ItemStack> getStack(@NotNull EquipmentSlot mcEquipmentSlot) {
-        var mcStack = getMinecraftEntity().getItemBySlot(mcEquipmentSlot);
-        if (mcStack == null || mcStack.isEmpty()) {
-            return Optional.empty();
-        }
-
-        return Optional.of(ItemStackImpl.of(mcStack));
+    public Optional<ItemStack> getEquipment(@NotNull EquipmentSlot slot) {
+        return Optional.ofNullable(getMinecraftEntity().getItemBySlot(getMinecraftEquipmentSlot(slot)))
+                .map(ItemStackImpl::of);
     }
 
     @Override
-    @NotNull
-    public Optional<ItemStack> getBoots() {
-        return getStack(EquipmentSlot.FEET);
+    public void setEquipment(@NotNull EquipmentSlot slot, @NotNull ItemStack equipment) {
+        getMinecraftEntity().setItemSlot(getMinecraftEquipmentSlot(slot), ((ItemStackImpl) equipment).getMinecraftItemStack());
     }
 
-    @Override
-    public void setBoots(@NotNull ItemStack itemStack) {
-        getMinecraftEntity().setItemSlot(EquipmentSlot.FEET, ((ItemStackImpl) itemStack).getMinecraftItemStack());
-    }
-
-    @Override
-    @NotNull
-    public Optional<ItemStack> getLeggings() {
-        return getStack(EquipmentSlot.LEGS);
-    }
-
-    @Override
-    public void setLeggings(@NotNull ItemStack itemStack) {
-        getMinecraftEntity().setItemSlot(EquipmentSlot.LEGS, ((ItemStackImpl) itemStack).getMinecraftItemStack());
-    }
-
-    @Override
-    @NotNull
-    public Optional<ItemStack> getChestplate() {
-        return getStack(EquipmentSlot.CHEST);
-    }
-
-    @Override
-    public void setChestplate(@NotNull ItemStack itemStack) {
-        getMinecraftEntity().setItemSlot(EquipmentSlot.CHEST, ((ItemStackImpl) itemStack).getMinecraftItemStack());
-    }
-
-    @Override
-    @NotNull
-    public Optional<ItemStack> getHelmet() {
-        return getStack(EquipmentSlot.HEAD);
-    }
-
-    @Override
-    public void setHelmet(@NotNull ItemStack itemStack) {
-        getMinecraftEntity().setItemSlot(EquipmentSlot.HEAD, ((ItemStackImpl) itemStack).getMinecraftItemStack());
+    private net.minecraft.world.entity.EquipmentSlot getMinecraftEquipmentSlot(@NotNull EquipmentSlot slot) {
+        return net.minecraft.world.entity.EquipmentSlot.values()[slot.ordinal()];
```

Comment:
Maybe a transformer for this couldn't hurt so we could get rid of the utility method in this class.